### PR TITLE
Add option to disable shuffling in datasplitter

### PR DESCRIPTION
**Pull request by martinkim0**

### CODE REVIEW

File: scvi/dataloaders/_data_splitting.py [hunk 1]
```diff
@@ -72,6 +72,8 @@ class DataSplitter(pl.LightningDataModule):
         float, or None (default is 0.9)
     validation_size
         float, or None (default is None)
+    shuffle
+        Whether to shuffle indices before splitting.
```
Comment (adamgayoso) [inline scvi/dataloaders/_data_splitting.py:76]:
```suggestion
        Whether to shuffle indices before splitting. If `False`, the val, train, and test set are split in the
        sequential order of the data according to `validation_size` and `train_size` percentages.
```

**Review by adamgayoso — APPROVED** [review]
LGTM after that one small change

File: scvi/dataloaders/_data_splitting.py [hunk 2]
```diff
@@ -95,13 +98,15 @@ def __init__(
         adata_manager: AnnDataManager,
         train_size: float = 0.9,
         validation_size: Optional[float] = None,
+        shuffle: bool = True,
```
Comment (adamgayoso) [inline scvi/dataloaders/_data_splitting.py:101]: actually can you expose this in unsupervisedtrainingmixn?

**Review by adamgayoso — CHANGES_REQUESTED** [review]

File: scvi/dataloaders/_data_splitting.py (hunk 2 above)
Comment (adamgayoso) [reply scvi/dataloaders/_data_splitting.py:101]: and totalvi.train, peakvi.train, multivi.train, JaxTrainingMixin.train, destvi.train, condscvi.train,

Comment (adamgayoso) [reply scvi/dataloaders/_data_splitting.py:101]: cellassign.train, scbasset.train, and then check solo and stereoscope

Comment (adamgayoso) [reply scvi/dataloaders/_data_splitting.py:101]: also could rename to `shuffle_set_split` as `shuffle` typically refers to minibatching

**Review by adamgayoso — CHANGES_REQUESTED** [review]
just need some tests then this LGTM

File: scvi/external/solo/_model.py [hunk 3]
```diff
@@ -334,6 +338,7 @@ def train(
             self.adata_manager,
             train_size=train_size,
             validation_size=validation_size,
+            shuffle=shuffle_set_split,
```
Comment (martinkim0) [inline scvi/external/solo/_model.py:341]:
```suggestion
            shuffle_set_split=shuffle_set_split,
```

**Review by adamgayoso — APPROVED** [review]

**Pull request merged**